### DISCUSSION TITLE
How to use way properties in relations?

### DISCUSSION THREAD

**Discussion by zerda-ocm** [onthegomap/planetiler#820] [Q&A] [answered]
I want to display highways at a lower zoom level than normal, if they are part of a route relation. For example "highway=path" is displayed at level 13 and higher, but when there is a hiking route on this path I want to display it at level 12 and higher.

I am currently using route relations like in this example: [BikeRouteOverlay.java](https://github.com/onthegomap/planetiler/blob/main/planetiler-examples/src/main/java/com/onthegomap/planetiler/examples/BikeRouteOverlay.java)

**Comment by bdon** [comment]
You need to use the first relations preprocessing step to store way information in memory, example: https://github.com/protomaps/basemaps/blob/main/tiles/src/main/java/com/protomaps/basemap/layers/Boundaries.java#L264

**Reply by zerda-ocm** [reply]
That's what I am already doing. But as far as I understand, this gives me the tags from the relation itself, but not the tags of the members of the relation.
Route relations itself do not have information about highway, surface, etc.
I basically want to split up a large relation into smaller relations with some extra tags I obtain from the ways that are part of the large relation.

**Reply by bdon** [reply]
store the IDs of the hiking relation on the first pass, and then apply the logic (highway=PATH AND ID is part of hiking relation) on the 2nd pass

**Reply by zerda-ocm** [reply, marked as answer]
`sourceFeature.getTag("highway")`
works for me.

```
public void processAllOsm(SourceFeature sourceFeature, FeatureCollector features) {
    // ignore nodes and ways that should only be treated as polygons
    if (sourceFeature.canBeLine()) {         
      for (var routeInfo : sourceFeature.relationInfo(RouteRelationInfo.class)) {  
        RouteRelationInfo relation = routeInfo.relation();
        String layerName = relation.route + "-route";
        features.line(layerName)
          .setAttr("name", relation.name)
          .setAttr("ref", relation.ref)
          .setAttr("route", relation.route)
          .setAttr("distance", relation.distance)
          .setAttr("osmc:symbol", relation.osmc_symbol)
          .setAttr("highway", sourceFeature.getTag("highway"))
          .setZoomRange(0, 14)
          .setMinPixelSize(0);
      }
    }
  }
```
